public/app/features/scenes/components/SceneBuilder.ts

```diff
@@ -0,0 +1,31 @@
+import { GraphFieldConfig, TableFieldOptions } from '@grafana/schema';
+import { PanelOptions as BarGaugePanelOptions } from 'app/plugins/panel/bargauge/models.gen';
+import { PanelOptions as TablePanelOptions } from 'app/plugins/panel/table/models.gen';
+import { TimeSeriesOptions } from 'app/plugins/panel/timeseries/types';
+
+import { VizPanel, VizPanelState } from './VizPanel/VizPanel';
+
+export type TypedVizPanelState<TOptions, TFieldConfig> = Omit<VizPanelState, 'pluginId'>;
+
+export class SceneBuilder {
+  public newGraph(state: TypedVizPanelState<TimeSeriesOptions, GraphFieldConfig>) {
+    return new VizPanel({
+      ...state,
+      pluginId: 'timeseries',
+    });
+  }
+
+  public newTable(state: TypedVizPanelState<TablePanelOptions, TableFieldOptions>) {
+    return new VizPanel<TablePanelOptions, TableFieldOptions>({
+      ...state,
+      pluginId: 'table',
+    });
+  }
+
+  public newBarGauge(state: TypedVizPanelState<BarGaugePanelOptions, {}>) {
+    return new VizPanel({
+      ...state,
+      pluginId: 'bargauge',
+    });
+  }
+}
```

Comment:
This is not used but could be an alternative to the typedPanels.newGraph , we would have all the core popular scene objects be created from functions on this class. 

The plus side would be discoverability (IntelliSense) and only having to import one type, but it's also a lot of duplicated constructor definitions to maintain. So opted not to use / go deeper into that, but left it here to see the remnant of something tried.